vernac/comFlags.ml

```diff
@@ -0,0 +1,235 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+open Goptions
+
+let allow_sprop_opt_name = ["Allow";"StrictProp"]
+let cumul_sprop_opt_name = ["Cumulative";"StrictProp"]
+
+let () =
+  declare_bool_option
+    { optdepr  = false;
+      optkey   = allow_sprop_opt_name;
+      optread  = (fun () -> Global.sprop_allowed());
+      optwrite = Global.set_allow_sprop }
+
+let () =
+  declare_bool_option
+    { optdepr  = false;
+      optkey   = cumul_sprop_opt_name;
+      optread  = Global.is_cumulative_sprop;
+      optwrite = Global.set_cumulative_sprop }
+
+let () =
+  declare_bool_option
+    { optdepr  = false;
+      optkey   = ["Silent"];
+      optread  = (fun () -> !Flags.quiet);
+      optwrite = ((:=) Flags.quiet) }
+
+let () =
+  declare_bool_option
+    { optdepr  = false;
+      optkey   = ["Implicit";"Arguments"];
+      optread  = Impargs.is_implicit_args;
```

Comment:
Shouldn't this go to impargs.ml instead?
AFAICT only the Global and Flags options need to be higher than the associated data

Reply:
Good point; on the other hand I'm trying to move `GOptions` to `vernac` so I dunno; we should maybe discuss about the design here a bit.

Reply:
Updated.

Reply:
I didn't mean just this specific option, there are a bunch more which shouldn't be in the new file.